src/commands/diff.ts

```diff
@@ -0,0 +1,109 @@
+import { flags } from '@oclif/command';
+import * as diff from '@asyncapi/diff';
+import * as parser from '@asyncapi/parser';
+import { load } from '../models/SpecificationFile';
+import Command from '../base';
+import { ValidationError } from '../errors/validation-error';
+import { SpecificationFileNotFound } from '../errors/specification-file';
+
+export default class Diff extends Command {
+  static description = 'find diff between two asyncapi files';
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+    format: flags.string({
+      char: 'o',
+      description: 'output format',
+      default: 'json',
+    }),
+    type: flags.string({
+      char: 't',
+      description: 'the type of output',
+      default: 'all',
+    }),
+  };
+
+  static args = [
+    {
+      name: 'first-spec-file',
+      description: 'first spec path or context-name',
+      required: true,
+    },
+    {
+      name: 'second-spec-file',
+      description: 'second spec path or context-name',
+      required: true,
+    },
+  ];
+
+  async run() {
+    const { args, flags } = this.parse(Diff);
+    const firstDocumentPath = args['first-spec-file'];
+    const secondDocumentPath = args['second-spec-file'];
+
+    const outputFormat = flags['format'];
+    const outputType = flags['type'];
+
+    let firstDocument, secondDocument;
+
+    try {
+      firstDocument = await load(firstDocumentPath);
+    } catch (err) {
+      if (err instanceof SpecificationFileNotFound) {
+        this.error(
+          new ValidationError({
+            type: 'invalid-file',
+            filepath: firstDocumentPath,
+          })
+        );
+      } else {
+        this.error(err as Error);
+      }
+    }
+
+    try {
+      secondDocument = await load(secondDocumentPath);
+    } catch (err) {
+      if (err instanceof SpecificationFileNotFound) {
+        this.error(
+          new ValidationError({
+            type: 'invalid-file',
+            filepath: secondDocumentPath,
+          })
+        );
+      } else {
+        this.error(err as Error);
+      }
+    }
+
+    try {
+      const firstDocumentParsed = await parser.parse(
+        await firstDocument.read()
+      );
+      const secondDocumentParsed = await parser.parse(
+        await secondDocument.read()
+      );
+      const diffOutput = diff.diff(
+        firstDocumentParsed.json(),
+        secondDocumentParsed.json()
+      );
+
+      if (outputFormat === 'json') {
```

Comment:
please have a look at this article https://thecodebarbarian.com/pretty-json-stringify-output.html

we should lot out json in a pretty formatted way and not block of text, it would be much nicer :)

Reply:
By autoloading context I mean by the feature when there is no context set but there is an asyncapi file in the working directory cli loads that right, but in this case where more than two spec file paths are required so that would not work as intended.

Reply:
ah, yeah, you mean fallback to `current` context. Yes, not a case here